### PR TITLE
provide storage_data_lake_gen2_filesystem  functionality

### LINKED ISSUE #156: Support for data lake gen2 filesystem
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Hi, it would be handy to include the option to make a storage account a data lake as well. 

### New or Affected Resource(s)/Data Source(s)

azurerm_storage_data_lake_gen2_filesystem

### Potential Terraform Configuration

```hcl
resource "azurerm_storage_data_lake_gen2_filesystem" "data_lakes_containers" {
  for_each = var.containers

  name               = each.value
  storage_account_id = azurerm_storage_account.data_lake_storage_account.id

  depends_on = [azurerm_role_assignment.data_lake_role_assignment]
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by chinthakaru**
## Description
provide storage_data_lake_gen2_filesystem  functionality

fixes #156 
## Type of Change

- [x] Non-module change (e.g. CI/CD, documentation, etc.)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates.
  - [ ] Breaking changes.
  - [x] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

### CODE REVIEW

File: variables.data_lake.tf [hunk 1]
```diff
@@ -0,0 +1,45 @@
+variable "storage_data_lake_gen2_filesystem" {
+  type = object({
+    default_encryption_scope = optional(string)
+    group                    = optional(string)
+    name                     = string
+    owner                    = optional(string)
+    properties               = optional(map(string))
+    #     storage_account_id       = string
```
Comment (jaredfholgate) [inline variables.data_lake.tf:8]: Can we remove this. Eventually this should be a submodule.

File: variables.data_lake.tf [hunk 2]
```diff
@@ -0,0 +1,45 @@
+variable "storage_data_lake_gen2_filesystem" {
+  type = object({
```
Comment (jaredfholgate) [inline variables.data_lake.tf:2]: Is this a 1-2-1, cannot specify more than 1 per storage account?

Comment (chinthakaru) [reply variables.data_lake.tf:2]: yes. This is file System deployment

**Review by jaredfholgate — APPROVED** [review]
LGTM

**Pull request merged**